### DISCUSSION TITLE
need help , No 'Access-Control-Allow-Origin' header is present on the requested resource problem

### DISCUSSION THREAD

**Discussion by xyychenchen** [shlinkio/shlink#2270] [Help wanted]
### Shlink version

4.2.2

### PHP version

8.2.25

### How do you serve Shlink

Self-hosted nginx

### Database engine

MariaDB

### Database version

10.11.6

### status quo 
I've been using shlink for days and it worked fine, suddenly today it appears the "Oops! Could not connect to this Shlink server.
Make sure you have internet connection, and the server is properly configured and on-line." like this 

![image](https://github.com/user-attachments/assets/ef2354cf-0dc7-4174-b5e9-eed9472b7e47)
 
And I have been checking the documents and using Chat Gpt to adjust the problem and try to fix it, accoridng to the instruction from AI I am right now adjusting my nginx reverse proxy configuration for shlink like this 
```
server {
    listen 443 ssl http2;
    server_name link.kkberry.com;
    
    root /var/www/html/shlink/public;
    index index.php;

    ssl_certificate /var/www/html/cert/kkberry.com.pem;
    ssl_certificate_key /var/www/html/cert/kkberry.com.key;
    
    ssl_session_cache shared:SSL:20m;
    ssl_session_timeout 10m;
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_prefer_server_ciphers on;
    ssl_ciphers 'ECDH+AESGCM:ECDH+AES256:ECDH+AES128:DH+3DES:!ADH:!AECDH:!MD5';

    # Security headers
    add_header X-Frame-Options "SAMEORIGIN";
    add_header X-Content-Type-Options "nosniff";
    add_header Strict-Transport-Security "max-age=31536000; includeSubDomains";
    add_header Referrer-Policy origin always;
    add_header X-XSS-Protection "1; mode=block" always;

    # CORS headers
    add_header 'Access-Control-Allow-Origin' '*';
    add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
    add_header 'Access-Control-Allow-Headers' 'Authorization, Content-Type';
    add_header 'Access-Control-Max-Age' 1728000;

    # Request handling
    location / {
        try_files $uri $uri/ /index.php?$query_string;
    }

    # PHP handling
    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass unix:/run/php/php8.2-fpm.sock;
        client_max_body_size 20m;
        fastcgi_connect_timeout 30s;
        fastcgi_send_timeout 30s;
        fastcgi_read_timeout 30s;
        fastcgi_intercept_errors on;
    }

    # Security settings
    location = /favicon.ico { access_log off; log_not_found off; }
    location = /robots.txt  { access_log off; log_not_found off; }
    error_page 404 /index.php;

    location ~ /\.(?!well-known).* { deny all; }
    location /tmp { deny all; }
    location ~ ^\. { deny all; }
    location ~ \.sqlite$ { deny all; }
    location ~ \.env$ { deny all; }
    location ~ /\.htaccess { allow all; }
}

```

I really run out of my knoledges can somebody help me a little bit, I have been talking to AI for the whole morning still cannot fix this

**Comment by xyychenchen** [comment]
no matter what I revised to the conf, it always shows the same error as the image display

**Comment by xyychenchen** [comment]
can you help me @acelaya thank you so much !

**Comment by xyychenchen** [comment]
I try to swith to a different browser, clear the cache in cloudflare it dont\  work

**Reply by SmartPhoneLover** [reply]
You're are using CF too, did you finally got it working?
I already reported this long long ago but never got an answer.

**Comment by xyychenchen** [comment]
I got it worked yes, it works randomly, you can let me know what did you encounter i can try to help

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
